### PR TITLE
Added ampere arch + impalajit as an external lib. (optionally)

### PR DESCRIPTION
**Pull request by ravil-mobile**
Added extra switches to CMake

### CODE REVIEW

**Comment by sebwolf-de** [comment]
Hi, I don't really understand the logic behind `IMPALAJIT_FROM_SUBMODULES`. 
I'd come up with the following logic (same as for yaml-cpp): 
* Try to find impalaJIT throught `find_package`
*  If not available (or wrong version) build from submodules.

**Comment by krenzland** [comment]
Yeah. I'd argue for the same. It would be easier to find a correct version if impalajit would support version numbers. Then you make sure that the version installed on the system is not outdated.
Burn considering that no one is adding features to impalajit anymore...

**Comment by ravil-mobile** [comment]
Hi, 

Thanks @sebwolf-de. I don't know why it didn't get into my mind and a better solution was a couple lines above. 

> if impalajit would support version numbers

It turns out that the original ImpalaJIT has a version. If you look at any of your SeisSol build directory you can find the following

```
$ cat ImpalaJIT-install/lib/pkgconfig/impalajit.pc
Name: ImpalaJIT
Description: JIT Compilation to realize flexible data access
URL: https://github.com/Manuel1605/ImpalaJIT
Version: 0.9

libdir=${prefix}/lib
includedir=${prefix}/include

Libs: -L${libdir} -limpalajit
Cflags: -I${includedir}
```

Version 1.0 corresponds to my fork of ImpalaJIT which I need to run DR on M100 and ARM+Nvidia nodes

**Comment by krenzland** [comment]
Could you maybe add a prefix/suffix to your version? E.g. llvm-1.0?
Otherwise this is going to make updating impalajit impossible

**Comment by ravil-mobile** [comment]
I can add cmake namespace or a component. Let me try. Btw, do we need to add it to docu or it is not necessary?

**Comment by ravil-mobile** [comment]
Now, it looks like the following

```
target_link_libraries(SeisSol-lib PRIVATE LLVM::impalajit)
```

Cmake macro `write_basic_package_version_file` requires only digits for a version. Here is an example, taken from a generated file:

```
et(PACKAGE_VERSION "1.0.0")

if(PACKAGE_VERSION VERSION_LESS PACKAGE_FIND_VERSION)
  set(PACKAGE_VERSION_COMPATIBLE FALSE)
else()

  if("1.0.0" MATCHES "^([0-9]+)\\.")
    set(CVF_VERSION_MAJOR "${CMAKE_MATCH_1}")
  else()
    set(CVF_VERSION_MAJOR "1.0.0")
  endif()

...
```

**Comment by krenzland** [comment]
I don't think that your approach solves the problem. It tries to find ANY ImpalaJIT with version 1.0 but then only links to LLVM::ImpalaJIT. If someone would publish ImpalaJIT with version 1.0 that does not use LLVM it wouldn't work.
Maybe you can rename your fork to ImpalaJIT-LLVM or similar? Then you can search for it.

**Comment by ravil-mobile** [comment]
Ok. I will rename the fork. Actually I only need to rename `ImpalaJITConfig.cmake` to `ImpalaJIT-LLVMConfig.cmake` (and corresponding files like version, etc). I wanted to do it at the beginning but I didn't know whether all of you would be ok with it

**Comment by krenzland** [comment]
Probably the best way  (in addition to renaming it to ImpalaJIT-LLVM or so) would be to add another compile time option to cmake, e.g. USE_IMPALA_JIT_LLVM or so. This would make it explicit and could avoid a lot of confusion. What do you think?

File: CMakeLists.txt [hunk 1]
```diff
@@ -157,20 +157,24 @@ else()
...
-)
-add_dependencies(SeisSol-lib ImpalaJIT-build)
-target_link_libraries(SeisSol-lib PUBLIC "${CMAKE_CURRENT_BINARY_DIR}/ImpalaJIT-install/lib/libimpalajit.a")
-target_include_directories(SeisSol-lib PUBLIC ${CMAKE_CURRENT_BINARY_DIR}/ImpalaJIT-install/include/)
+if (USE_IMPALA_JIT_LLVM)
+  find_package(ImpalaJIT-LLVM 1.0 REQUIRED)
+  target_link_libraries(SeisSol-lib PRIVATE llvm::impalajit)
+else()
+  ExternalProject_Add(ImpalaJIT-build
+    SOURCE_DIR ${CMAKE_CURRENT_LIST_DIR}/submodules/ImpalaJIT/
```
Comment (ravil-mobile) [inline CMakeLists.txt:165]: Is it OK now?

**Review by krenzland — APPROVED** [review]

**Pull request merged**